File: Python/Samples/State/UtState.py
import unittest
import Context


class UtState(unittest.TestCase):

    def test_state(self):
        expectedFinalState = "Done(已完成)"
        actualFinalState = ""

        context = Context.Context()
        while (context.currentState != None):
            actualFinalState = str(context.currentState)
            print("需求目前狀態="+ actualFinalState)
            context.action()
            
        self.assertEqual(actualFinalState, expectedFinalState)



if __name__ == '__main__':
    unittest.main()

File: CSharp/DP.UnitTest/UtState.cs
using System;
using System.Diagnostics;
using DP.Domain.Samples.State;
using DP.UnitTest.Utility;
using Xunit;
using Xunit.Abstractions;

namespace DP.UnitTest
{
    public class UtState
    {
        private readonly ITestOutputHelper _output;

        public UtState(ITestOutputHelper output)
        {
            this._output = output;
            System.Diagnostics.Trace.Listeners.Add(new XunitTraceListener(this._output));
        }

        [Fact]
        public void TestState()
        {
            var expectedFinalState = "Done(已完成)";
            var actualFinalState = string.Empty;

            var context = new Context();
            while (context.CurrentState != null)
            {
                actualFinalState = context.CurrentState.ToString();
                Trace.WriteLine($"需求目前狀態={actualFinalState}");
                context.Action();
            }

            Assert.Equal(expectedFinalState, actualFinalState);
        }
    }
}
